src/reference/asciidoc/kafka.adoc

```diff
@@ -2357,6 +2392,7 @@ For more complex or particular cases, the `KafkaConsumer` (and, therefore, `Kafk
 constructors to accept `Serializer` and `Deserializer` instances for `keys` and `values`, respectively.
 
 When you use this API, the `DefaultKafkaProducerFactory` and `DefaultKafkaConsumerFactory` also provide properties (through constructors or setter methods) to inject custom `Serializer` and `Deserializer` instances into the target `Producer` or `Consumer`.
+Also, you can pass in `Supplier` functions for custom `Serializer` and `Deserializer` instances through constructors - these functions are called on creation of each `Producer` or `Consumer`.
```

Comment:
`Supplier` is not a `Function`. See my comment below about generics opportunity.